lib/galaxy/tools/errors.py

```diff
@@ -51,6 +52,91 @@
 (This is an automated message).
 """
 
+error_report_template_html = """
+<html><head></head>
+    <body>
+<style type="text/css">
+tr:nth-child(even) {background-color: #f2f2f2}
+table{margin: 1em;}
+.mono{font-family: monospace;}
+pre {
+white-space: pre-wrap;
+white-space: -moz-pre-wrap;
+white-space: -pre-wrap;
+white-space: -o-pre-wrap;
+word-wrap: break-word;
+background: #eee;
+border:1px solid black;
+padding:10px;
+}
+</style>
+
+<h1>Galaxy Tool Error Report</h1>
+<span class="sub"><i>from</i> <span class="mono"><a href="${host}">${host}</a></span>
+
+<h3>Error Localization</h3>
+<table>
+    <tbody>
+        <tr><td>Dataset</td><td>${dataset_id} (${dataset_id_encoded})</td></tr>
+        <tr><td>History</td><td><a href="${history_view_link}">${history_id} (${history_id_encoded})</a></td></tr>
+        <tr><td>Failed Job</td><td>${hid}: ${history_item_name}</td></tr>
```

Comment:
Because the TXT report wasn't consistent at all. Had I spotted that I would've added it.

Txt report:

> reference to dataset id ${dataset_id} (${dataset_id_encoded}) from history id ${history_id} (${history_id_encoded}) ... history containing the related history item (${hda_id_encoded})

But yes, I can patch this.